src/controllers/controller.bar.js

```diff
@@ -4,13 +4,14 @@ import {
   valueOrDefault, resolveObjectKey, sign, defined
 } from '../helpers';
 
-function getAllScaleValues(scale) {
+function getAllScaleValues(meta) {
+  let scale = meta.iScale;
   if (!scale._cache.$bar) {
-    const metas = scale.getMatchingVisibleMetas('bar');
+    const visibleMetas = scale.getMatchingVisibleMetas(meta._type);
```

Comment:
should be `meta.type` (or `datasetController._type`)

I'd keep the `scale` parameter, as the function is called `getAllScaleValues` and add `type` as 2nd argument.

Reply:
Yes. That will be more appropriate to what function `getAllScaleValues` is performing